composer.json

```diff
@@ -4,12 +4,13 @@
     "type": "cakephp-plugin",
     "license": "MIT",
     "require": {
-        "cakephp/cakephp": "^3.6",
-        "spiral/roadrunner": "^1.3",
-        "relay/middleware": "^1.2"
+        "php": "^8.0",
+        "cakephp/cakephp": "^4.3",
+        "spiral/roadrunner": "^2.7",
+        "nyholm/psr7": "^1.4"
```

Comment:
Why is this package needed? Cake already has PSR-7 complaint classes.

Reply:
Good catch, its from the roadrunner examples.

Reply:
You'll probably be able to get rid of the `convertRequest()` method too from `Bridge` if you pass Cake's `ServerRequest` from the worker file.

Reply:
I'm not seeing Cake implementations of the following:

- Psr\Http\Message\StreamFactoryInterface
- Psr\Http\Message\UploadedFileFactoryInterface

Reply:
We use laminas/diactoros' classes for those interfaces.

Is explicitly creating instances for those even required? It's handled internally in the `ServerRequest` class.

Reply:
Thanks, thats resolved now.

Reply:
Hmm... except it no longer works when those are replaced!

Reply:
Maybe the environment vars it sets (using `$server`) need to be set explicitly. Try to narrow down what's the relevant part it does and then directly modify the `$request` instance. Recreating the request instance again surely shouldn't be necessary if it's already a `Cake\Http\ServerRequest` instance.

Reply:
Hmmm.... I had a full comment on this that disappeared. Anyways I added a doc block to `convertRequest` that explains the need for it (for now). I cleaned it up though. Basically cake internals and some plugin middleware need an instance of Cakes ServerRequest.

Reply:
Using `Cake\Http\ServerRequestFactory` instead of `Laminas\Diactoros\ServerRequestFactory` in the worker file would avoid this problem and remove the need for `Bridge::convertRequest()` too.

But for some reason `Cake\Http\ServerRequestFactory` is made `abstract` so it can't be instantiated. No idea why that is. Perhaps something we should change in the core.

Reply:
So this is why `CakeDC\Roadrunner\Http\ServerRequestFactory` was required and you should use that in the worker file :slightly_smiling_face: